sisl/geom/nanoribbon.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+
+from sisl import geom, Atom, Geometry
+
+__all__ = ['nanoribbon', 'graphene_nanoribbon', 'agnr', 'zgnr']
+
+
+def nanoribbon(bond, atom, width, kind='armchair'):
+    r""" Construction of a nanoribbon unit cell of type armchair or zigzag.
+    The geometry is oriented along the x-axis.
+
+    Parameters
+    ----------
+    bond : float
+       bond length between atoms in the honeycomb lattice
+    atom : Atom
+       atom (or atoms) in the honeycomb lattice
+    width : int
+       number of atoms in the transverse direction
+    kind : {'armchair', 'zigzag'}
+       type of ribbon
+
+    See Also
+    --------
+    honeycomb : honeycomb lattices
+    graphene : graphene geometry
+    graphene_nanoribbon : graphene nanoribbon
+    agnr : armchair graphene nanoribbon
+    zgnr : zigzag graphene nanoribbon
+    """
+    if not isinstance(bond, float):
```

Comment:
Users *could* use integers, generally I wouldn't recommend doing any checks on this, it will be caught later if needed. If you really want to check, a better check would be something like this:
```python
from numbers import Real

if isinstance(bond, Real):
    ...
```
the `Real` matches both integers and floats (but not complex values) :)